src/components/YoutubePlayer/YoutubePlayer.js

```diff
@@ -0,0 +1,122 @@
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import {
+  faThumbsUp,
+  faThumbsDown,
+  faClock,
+  faIndent,
+} from "@fortawesome/free-solid-svg-icons";
+
+import React, { useState } from "react";
+import YouTube from "react-youtube";
+import { useVideo } from "../../contexts/video-context";
+import "./youtubePlayer.css";
+import { PlayListAdd } from "./../PlayListAdd/PlayListAdd";
+import { useLocation, useParams } from "react-router-dom";
+export function YoutubePlayer() {
+  const [playListWindow, setPlaylistWindow] = useState(false);
+  const [sizeOfWindow, setSizeOfWindow] = useState(window.innerWidth);
+  const { state, dispatch, videoList } = useVideo();
+  const opts = {
+    height: sizeOfWindow > 900 ? "550" : "300",
+    width: "100%",
+    playerVars: {
+      autoplay: 1,
+    },
+  };
+
+  window.onresize = () => {
+    setSizeOfWindow(window.innerWidth);
+  };
+
+  const query = new URLSearchParams(useLocation().search);
+
+  const video = videoList.find((item) => item.id === query.get("id"));
+
+  function handleWatchLater(video) {
+    return !!state.watchLater.find((item) => item.id === video.id)
+      ? null
+      : dispatch({ type: "ADD_TO_WATCH_LATER", video });
+  }
+
```

Comment:
what is this double ! for? i don't know.. do let me know

Reply:
well it is to convert the truthy value to defined true or false value